src/main/java/com/github/bibsysdev/nva/cristin/person/CristinApiClient.java

```diff
@@ -0,0 +1,190 @@
+    private static final String SEARCH_PATH = "search?QUERY_PARAMS"; // TODO: NP-2412: Replace QUERY_PARAMS
+    private static final String ERROR_MESSAGE_FETCHING_CRISTIN_PERSON_WITH_ID = "Error fetching cristin person with "
+        + "id: %s . Exception Message: %s";
+    private static final String ERROR_MESSAGE_BACKEND_FETCH_FAILED = "Your request cannot be processed at this time "
+        + "due to an upstream error";
+
+    /**
+     * Creates a wrapper object containing CristinPersons transformed to NvaPersons. Is used for serialization to the
+     * client.
+     *
+     * @param parameters The query params
+     * @param language   Language used for some properties in Cristin API response
+     * @return a PersonsWrapper with transformed CristinPersons
+     * @throws IOException        for connection errors
+     * @throws URISyntaxException if URI is malformed
+     */
+    public PersonsWrapper queryCristinPersonsIntoWrapperObject(Map<String, String> parameters, String language)
+        throws IOException, URISyntaxException {
+
+        long startRequestTime = System.currentTimeMillis();
+        List<CristinPerson> enrichedPersons = queryAndEnrichPersons(parameters, language);
+        long endRequestTime = System.currentTimeMillis();
+
+        PersonsWrapper personsWrapper = new PersonsWrapper();
+        personsWrapper.setId(buildUri(BASE_URL, SEARCH_PATH));
+        personsWrapper.setSize(0); // TODO: NP-2385: X-Total-Count header from Cristin response
+        personsWrapper.setSearchString(extractNameSearchString(parameters));
+        personsWrapper.setProcessingTime(calculateProcessingTime(startRequestTime, endRequestTime));
+        // TODO: NP-2385: Use Link header / Pagination data from Cristin response in the next two values
+        personsWrapper.setFirstRecord(0);
+        personsWrapper.setNextResults(null); // TODO: Change to URI
+        personsWrapper.setHits(transformCristinPersonsToNvaPersons(enrichedPersons));
+        // TODO: NP-2424: Return fields with empty values instead of null to avoid "undefined" in frontend
+        return personsWrapper;
+    }
+
+    /**
+     * Creates a NvaPerson object containing a single transformed CristinPerson. Is used for serialization to the
+     * client.
+     *
+     * @param id       The CristinPersonId of the person to query
+     * @param language Language used for some properties in the Cristin API response
+     * @return a NvaPerson filled with one transformed CristinPerson
+     * @throws BadGatewayException when there is a problem with fetch from backend
+     */
+    public NvaPerson queryOneCristinPerson(String id, String language) throws BadGatewayException {
+        CristinPerson cristinPerson = attemptToGetCristinPerson(id, language);
+        if (cristinPerson == null || !cristinPerson.hasRequiredFields()) {
+            return new EmptyNvaPerson();
+        } else {
+            NvaPerson nvaPerson = new NvaPersonBuilder(cristinPerson).build();
```

Comment:
I would suggest to put the command new NvaPresionBuilder(cristinPerson).build() in static mehtod in NvaPerson::fromCristinPerson.
```
public static NvaPersion fromCristinPerson(CristinPersion cp){
  return new NvaPersonBuilder(cristinPerson).build();
}

Reply:
I don't mind, but for my elucidation; how does this not deviate from the builder pattern? Would making the entire builder class a static class within the person class be a better/worse/acceptable alternative? Does this interfere with the workings of jackson any?

Reply:
My understanding of the builder pattern is that you only call the builder from a director, never from the class to be built. I don't mind doing it a different way, it's just new to me.

Reply:
The builder should be able to be located from the class to be built.Otherwise, how would you know that there is a builder?
1. https://dzone.com/articles/design-patterns-the-builder-pattern
2. https://howtodoinjava.com/design-patterns/creational/builder-pattern-in-java/
3. An example of an SDK with many builders: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3Client.html

Reply:
Your first two examples do not seem to use the suggested change, but rather includes the builder in the model class. V1 of the sdk you linked does seem to do something similar though. I will enact the change and read up on it. Thank you.